pkg/dss/scd/cockroach/store.go

```diff
@@ -0,0 +1,124 @@
+package cockroach
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/dpjacques/clockwork"
+	"go.uber.org/zap"
+)
+
+var (
+	// DefaultClock is what is used as the Store's clock, returned from Dial.
+	DefaultClock = clockwork.NewRealClock()
+)
+
+// Store is an implementation of dss.Store using
+// Cockroach DB as its backend store.
+type Store struct {
+	*sql.DB
+	logger *zap.Logger
+	clock  clockwork.Clock
+}
+
+// Dial returns a Store instance connected to a cockroach instance available at
+// "uri".
+// https://www.cockroachlabs.com/docs/stable/connection-parameters.html
+func Dial(uri string, logger *zap.Logger) (*Store, error) {
+	db, err := sql.Open("postgres", uri)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Store{
+		DB:     db,
+		logger: logger,
+		clock:  DefaultClock,
+	}, nil
+}
+
+// BuildURI returns a cockroachdb connection string from a params map.
+func BuildURI(params map[string]string) (string, error) {
+	an := params["application_name"]
+	if an == "" {
+		an = "dss"
+	}
+	h := params["host"]
+	if h == "" {
+		return "", errors.New("missing crdb hostname")
+	}
+	p := params["port"]
+	if p == "" {
+		return "", errors.New("missing crdb port")
+	}
+	u := params["user"]
+	if u == "" {
+		return "", errors.New("missing crdb user")
+	}
+	ssl := params["ssl_mode"]
+	if ssl == "" {
+		return "", errors.New("missing crdb ssl_mode")
+	}
+	if ssl == "disable" {
+		return fmt.Sprintf("postgresql://%s@%s:%s?application_name=%s&sslmode=disable", u, h, p, an), nil
+	}
+	dir := params["ssl_dir"]
+	if dir == "" {
+		return "", errors.New("missing crdb ssl_dir")
+	}
+
+	return fmt.Sprintf(
+		"postgresql://%s@%s:%s?application_name=%s&sslmode=%s&sslrootcert=%s/ca.crt&sslcert=%s/client.%s.crt&sslkey=%s/client.%s.key",
+		u, h, p, an, ssl, dir, dir, u, dir, u,
+	), nil
+}
+
+type queryable interface {
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
+	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+}
+
+// Close closes the underlying DB connection.
+func (s *Store) Close() error {
+	return s.DB.Close()
+}
+
+// Bootstrap bootstraps the underlying database with required tables.
+//
+// TODO: We should handle database migrations properly, but bootstrap both us
+// *and* the database with this manual approach here.
+func (s *Store) Bootstrap(ctx context.Context) error {
+	const query = `
+	CREATE TABLE IF NOT EXISTS scd_subscriptions (
+		id UUID PRIMARY KEY,
+		owner STRING NOT NULL,
+		version INT4 NOT NULL DEFAULT 0,
+		url STRING NOT NULL,
+		notification_index INT4 DEFAULT 0,
+		notify_for_operations BOOL DEFAULT false,
```

Comment:
super nit: "include_operations" makes me think less than "notify_for_..."

Reply:
This is based on [the API](https://github.com/astm-utm/Protocol/blob/d7e55fff0b4c55203c20e9408643f20d1cd6435a/utm.yaml#L503) -- a Subscription with this flag set will trigger notifications for changes to Operations in the airspace.

Reply:
Given nit status, I'll push this until later if we want to open an issue.